File: l2-domain/src/main/java/com/vvygulyarniy/l2/domain/skill/SkillType.java
package com.vvygulyarniy.l2.domain.skill;

/**
 * Created by Phoen-X on 25.02.2017.
 */
public enum SkillType {
    CLASS,
    FISHING,
    PLEDGE,
    SUBPLEDGE,
    TRANSFORM,
    TRANSFER,
    SUBCLASS,
    COLLECT;

    public static SkillType getSkillType(int id) {
        return values()[id];
    }
}

File: l2-domain/src/main/java/com/vvygulyarniy/l2/domain/skill/SkillType.kt
package com.vvygulyarniy.l2.domain.skill

enum class SkillType {
    CLASS,
    FISHING,
    PLEDGE,
    SUBPLEDGE,
    TRANSFORM,
    TRANSFER,
    SUBCLASS,
    COLLECT;


    companion object {
        @JvmStatic fun getSkillType(id: Int): SkillType {
            return values()[id]
        }
    }
}
